src/libData/BlockData/BlockHeader/FallbackBlockHeader.cpp

```diff
@@ -0,0 +1,208 @@
+* Some programs in this code are governed by the GNU General Public License v3.0 (available at 
+* https://www.gnu.org/licenses/gpl-3.0.en.html) (‘GPLv3’). The programs that are governed by 
+* GPLv3.0 are those programs that are located in the folders src/depends and tests/depends 
+* and which include a reference to GPLv3 in their program files.
+**/
+
+#include "FallbackBlockHeader.h"
+#include "libUtils/Logger.h"
+
+using namespace std;
+using namespace boost::multiprecision;
+
+FallbackBlockHeader::FallbackBlockHeader()
+    : m_leaderConsensusId(0)
+{
+}
+
+FallbackBlockHeader::FallbackBlockHeader(const vector<unsigned char>& src,
+                                         unsigned int offset)
+{
+    if (Deserialize(src, offset) != 0)
+    {
+        LOG_GENERAL(INFO,
+                    "Error. We failed to initialize FallbackBlockHeader.");
+    }
+}
+
+FallbackBlockHeader::FallbackBlockHeader(
+    const uint64_t& fallbackDSEpochNo, const uint64_t& fallbackEpochNo,
+    const unsigned char fallbackState, const StateHash& stateRootHash,
+    const uint32_t leaderConsensusId, const Peer& leaderNetworkInfo,
+    const PubKey& leaderPubKey, const uint32_t shardId,
+    const boost::multiprecision::uint256_t& timestamp)
+    : m_fallbackDSEpochNo(fallbackDSEpochNo)
+    , m_fallbackEpochNo(fallbackEpochNo)
+    , m_fallbackState(fallbackState)
+    , m_hash{stateRootHash}
+    , m_leaderConsensusId(leaderConsensusId)
+    , m_leaderNetworkInfo(leaderNetworkInfo)
+    , m_leaderPubKey(leaderPubKey)
+    , m_shardId(shardId)
+    , m_timestamp(timestamp)
+{
+}
+
+unsigned int FallbackBlockHeader::Serialize(vector<unsigned char>& dst,
+                                            unsigned int offset) const
+{
+    unsigned int size_needed = FallbackBlockHeader::SIZE;
+    unsigned int size_remaining = dst.size() - offset;
+
+    if (size_remaining < size_needed)
+    {
+        dst.resize(size_needed + offset);
+    }
+
+    unsigned int curOffset = offset;
+    SetNumber<uint64_t>(dst, curOffset, m_fallbackDSEpochNo, sizeof(uint64_t));
+    curOffset += sizeof(uint64_t);
+    SetNumber<uint64_t>(dst, curOffset, m_fallbackEpochNo, sizeof(uint64_t));
+    curOffset += sizeof(uint64_t);
+    SetNumber<unsigned char>(dst, curOffset, m_fallbackState,
+                             sizeof(unsigned char));
+    curOffset += sizeof(unsigned char);
+    curOffset = m_hash.Serialize(dst, curOffset);
+    SetNumber<uint32_t>(dst, curOffset, m_leaderConsensusId, sizeof(uint32_t));
+    curOffset += sizeof(uint32_t);
+    curOffset += m_leaderNetworkInfo.Serialize(dst, curOffset);
+    curOffset += m_leaderPubKey.Serialize(dst, curOffset);
+    SetNumber<uint32_t>(dst, curOffset, m_shardId, sizeof(uint32_t));
+    curOffset += sizeof(uint32_t);
+    SetNumber<uint256_t>(dst, curOffset, m_timestamp, UINT256_SIZE);
+    curOffset += UINT256_SIZE;
+    return size_needed;
+}
+
+int FallbackBlockHeader::Deserialize(const vector<unsigned char>& src,
```

Comment:
Should not you use protobuf ?

Reply:
It's used, but the old school serialization is also implemented

Reply:
Any example to refer to?

Reply:
perhaps @ansnunez can guide what is correct

Reply:
Ah never mind I see you actually already implemented and used protobuf